src/backend/cuda/CMakeLists.txt

```diff
@@ -104,7 +103,7 @@ IF(UNIX)
         # Enabling c++11 with nvcc 7.5 + gcc 6.x doesn't seem to work
         # Only solution for now is to force use c++03 for gcc 6.x
         IF(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "6.0.0")
-          SET(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -Xcompiler -std=c++98")
+          message( FATAL_ERROR "NVCC does not support GCC version 6.0 or greater." )
```

Comment:
But is this condition really required? I know CUDA applies the same restrictions in `host_config.h`, but that would mean we would have to update CMake everytime CUDA updates their max gcc version.